Source/Dafny/Compiler-java.cs

```diff
@@ -720,7 +720,13 @@ protected class ClassWriter : IClassWriter {
       } else if (AsNativeType(e.Type) != null) {
         GetNativeInfo(AsNativeType(e.Type).Sel, out var name, out var literalSuffix, out _);
         var cast = name == "Short" || name == "Byte" ? $"({name.ToLower()})" : "";
-        wr.Write($"new {name}({cast}{(BigInteger)e.Value}{literalSuffix})");
+        var intValue = (BigInteger)e.Value;
+        if (intValue > long.MaxValue) {
+          // Represent the value as a signed 64-bit integer, which the ULong
```

Comment:
What prevents `intValue` from being gigantic, so that `intValue - ulong.MaxValue - 1` is still too big for the cast?

Reply:
Only the fact that the number has a native type, so it's at most a 64-bit unsigned integer.  (Probably worth a comment and/or assertion?)